source/extensions/filters/http/ext_proc/processor_state.cc

```diff
@@ -456,7 +456,7 @@ absl::Status ProcessorState::handleTrailersResponse(const TrailersResponse& resp
       bodyMode() == ProcessingMode::FULL_DUPLEX_STREAMED) {
     ENVOY_LOG(debug, "Applying response to buffered trailers, body_mode_ {}",
               ProcessingMode::BodySendMode_Name(body_mode_));
-    if (response.has_header_mutation()) {
+    if (response.has_header_mutation() && trailers_ != nullptr) {
```

Comment:
It looks to me this is the issue. To simply fix the issue, you can just add the check like below to fix the fuzzer issue:
if (response.has_header_mutation() && trailers_available_)

However, it might be a good idea to remove the trailers_available_ flag and directly check (trailers_ == nullptr) as proposed in the PR. These two are equivalent anyway based on the code logic.